patches/port_test_api.patch

```diff
@@ -0,0 +1,396 @@
++{
++  void* p1 = mi_malloc_aligned(48,32);
++	UK_TEST_EXPECT(p1 != NULL);
++  UK_TEST_EXPECT((uintptr_t)(p1) % 32 == 0);
++
++  void* p2 = mi_malloc_aligned(48,32);
++	UK_TEST_EXPECT(p2 != NULL);
++  UK_TEST_EXPECT((uintptr_t)(p2) % 32 == 0);
++
++  mi_free(p2);
++  mi_free(p1);
++}
++
++UK_TESTCASE(libmimalloc_api_testsuite, malloc_aligned4)
++{
++  void* p;
++  bool ok = true;
++  for (int i = 0; i < 8 && ok; i++) {
++    p = mi_malloc_aligned(8, 16);
++    ok = (p != NULL && (uintptr_t)(p) % 16 == 0); mi_free(p);
++  }
++  UK_TEST_EXPECT(ok);
++}
++
++UK_TESTCASE(libmimalloc_api_testsuite, malloc_aligned_at1)
++{
++  void* p = mi_malloc_aligned_at(48,32,0);
++  UK_TEST_EXPECT(p != NULL);
++	UK_TEST_EXPECT(((uintptr_t)(p) + 0) % 32 == 0);
++  mi_free(p);
++}
++
++UK_TESTCASE(libmimalloc_api_testsuite, malloc_aligned_at2)
++{
++  void* p = mi_malloc_aligned_at(50,32,8);
++  UK_TEST_EXPECT(p != NULL);
++	UK_TEST_EXPECT(((uintptr_t)(p) + 8) % 32 == 0);
++  mi_free(p);
++}
++
++UK_TESTCASE(libmimalloc_api_testsuite, memalign1)
++{
++  void* p;
++  bool ok = true;
++  for (int i = 0; i < 8 && ok; i++) {
++    p = mi_memalign(16,8);
++    ok = (p != NULL && (uintptr_t)(p) % 16 == 0); mi_free(p);
++  }
++  UK_TEST_EXPECT(ok);
++}
++
++// ---------------------------------------------------
++// Heaps
++// ---------------------------------------------------
++UK_TESTCASE(libmimalloc_api_testsuite, heap_destroy)
++{
+   mi_heap_t* heap = mi_heap_new();
+   int* p1 = mi_heap_malloc_tp(heap,int);
+   int* p2 = mi_heap_malloc_tp(heap,int);
+   *p1 = *p2 = 43;
+   mi_heap_destroy(heap);
+-  return true;
+ }
+ 
+-bool test_heap2() {
++UK_TESTCASE(libmimalloc_api_testsuite, heap_delete)
++{
+   mi_heap_t* heap = mi_heap_new();
+   int* p1 = mi_heap_malloc_tp(heap,int);
+   int* p2 = mi_heap_malloc_tp(heap,int);
+@@ -218,29 +203,39 @@ bool test_heap2() {
+   *p1 = 42;
+   mi_free(p1);
+   mi_free(p2);
+-  return true;
+ }
+ 
+-bool test_stl_allocator1() {
++// ---------------------------------------------------
++// various
++// ---------------------------------------------------
++UK_TESTCASE(libmimalloc_api_testsuite, realpath)
++{
++  char* s = mi_realpath( ".", NULL );
++  mi_free(s);
++}
++
+ #ifdef __cplusplus
++UK_TESTCASE(libmimalloc_api_testsuite, stl_allocator1)
++{
+   std::vector<int, mi_stl_allocator<int> > vec;
+   vec.push_back(1);
+   vec.pop_back();
+-  return vec.size() == 0;
+-#else
+-  return true;
+-#endif
++  UK_TEST_EXPECT_ZERO(vec.size());
+ }
++#endif
+ 
++#ifdef __cplusplus
+ struct some_struct  { int i; int j; double z; };
+ 
+-bool test_stl_allocator2() {
+-#ifdef __cplusplus
++UK_TESTCASE(libmimalloc_api_testsuite, stl_allocator2)
++{
+   std::vector<some_struct, mi_stl_allocator<some_struct> > vec;
+   vec.push_back(some_struct());
+   vec.pop_back();
+-  return vec.size() == 0;
+-#else
+-  return true;
+-#endif
++  UK_TEST_EXPECT_ZERO(vec.size());
+ }
++#endif
++
++#if defined(CONFIG_LIBMIMALLOC_TEST_API)
```

Comment:
See the comment on the `Config.uk` file, what is the purpose of `LIBMIMALLOC_TEST_API`?

Reply:
Updated, In case User's want to enable/disable certain testsuites, for now it only support one, but the flag is there for the future.